packages/0xcert-ethereum-generic-provider/src/core/mutation.ts

```diff
@@ -239,6 +239,77 @@ export class Mutation extends EventEmitter implements MutationBase {
     return this.resolveCurrentState();
   }
 
+  /**
+   * Retries mutation with higher gas price if possible. It uses providers retryGasMultiplier to
+   * calculate gas price.
+   * @notice Return error if mutation is already accepted into the Blockchain.
+   */
+  public async retry() {
+    const tx = await this.getTransactionObject();
+    if (!tx) {
+      throw new Error('Mutation not found');
+    } else if (tx.blockNumber) {
+      throw new Error('Mutation already acceped into the Blockchain');
+    }
+
+    const gasPrice = await this._provider.post({
+      method: 'eth_gasPrice',
+      params: [],
+    });
+    const oldGasPrice = tx.gasPrice;
+    const retryGasPrice = gasPrice.result * this._provider.retryGasPriceMultiplier;
+    // We first calculate new gas price based on current network conditions and
+    // retryGasPriceMultiplier if calculated gas price is lower then the original gas price then we
+    // use the retryGasPriceMultiplier on the original gas price.
+    const newGasPrice = retryGasPrice >= oldGasPrice ? retryGasPrice : oldGasPrice * this._provider.retryGasPriceMultiplier;
+    const attrs = {
+      from: tx.from,
+      data: tx.input,
+      nonce: tx.nonce,
+      value: tx.value,
+      gas: tx.gas,
+      gasPrice: `0x${Math.ceil(newGasPrice).toString(16)}`,
+    };
+    if (tx.to) {
+      attrs['to'] = tx.to;
+    }
+    const res = await this._provider.post({
+      method: 'eth_sendTransaction',
+      params: [attrs],
+    });
+    this._id = res.result;
+  }
+
+  /**
+   * Cancels mutation if possible.
+   * @notice Return error if mutation is already accepted into the Blockchain or you are not the
```

Comment:
* @notice Returns error if a mutation is already accepted onto the blockchain or if you are not the